services/web-app/utils/schema.js

```diff
@@ -20,11 +20,7 @@ import { type } from '@/data/type'
 export const statusSortComparator = (assetA, assetB) => {
   const statusKeys = Object.keys(status)
 
-  const statusA =
-    typeof assetA.content.status === 'object' ? assetA.content.status.key : assetA.content.status
-  const statusB =
-    typeof assetB.content.status === 'object' ? assetB.content.status.key : assetB.content.status
-  return statusKeys.indexOf(statusA) > statusKeys.indexOf(statusB) ? 1 : -1
+  return statusKeys.indexOf(assetA.statusKey) > statusKeys.indexOf(assetB.statusKey) ? 1 : -1
```

Comment:
Unfortunately, this might not behave in the way you'd expect. The iteration order is always the same, but it's not necessarily the insertion order that this is relying on.

https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order

Also, if this gets transpiled to something ES2015 or older, there's no guarantee on the property order at all.

A better approach to achieve the intended behavior would be to store the status objects in an array where the `key` is one of the properties in each object, or in a `Map`.

You could also keep the object as-is, but have a second var in the `status.js` file which is an array of keys in the well-defined order.

Reply:
added `assetStatusLifecycle` array